mantidproject/mantid
# Added masking-by-component to MaskDetectors

This PR adds a new *ComponentList* input property to [`MaskDetectors`](http://docs.mantidproject.org/nightly/algorithms/MaskDetectors-v1.html). The property takes a list of component names as strings and adds the detectors found recursively in the given components to the list of detector IDs to be masked.

Additionally, `ComponentCreationHelper::createTestInstrumentCylindrical()` was modified slightly to enable proper unit testing and the affected `InstrumentRayTracerTest` was updated.

**To test:**

Load your favorite dataset with attached instrument or just an empty instrument using [`LoadEmptyInstrument`](http://docs.mantidproject.org/nightly/algorithms/LoadEmptyInstrument-v1.html). Try masking different components in the component tree.
- Using a plain component name should mask all components with the name.
- Slash '/' can be used to more accurately specify a component, e.g. 'bank1/tube3'. However, in this case only a single component whose name matches the component string is masked.
- The algorithm should work regardless of the combination of parameters given, e.g. one is able to specify components, spectrum numbers and limit the masked range by *StartWorkspaceIndex* and *EndWorkspaceIndex* simultaneously.
- Non-existing component names don't fail but should print an error to the log.
- Component names with commas ',' will just confuse the algorithm. Consider updating your IDFs.

Fixes #18327 

*Release notes were updated accordingly.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
